client/state/jetpack-connect/actions.js

```diff
@@ -195,6 +195,64 @@ export function retryAuth( url, attemptNumber ) {
 	};
 }
 
+/**
+ * Create a user account
+ *
+ * !! Must have same return shape as createAccount !!
+ *
+ * @param  {Object} socialInfo              …
+ * @param  {Object} socialInfo.service      The name of the social service
+ * @param  {Object} socialInfo.access_token An OAuth2 acccess token
+ * @param  {Object} socialInfo.id_token     (Optional) a JWT id_token which contains the signed user info
+ *
+ * @return {Promise}                  Resolves to { username, bearerToken }
+ */
+export function createSocialAccount( socialInfo ) {
+	return async dispatch => {
+		dispatch( recordTracksEvent( 'calypso_jpc_social_createaccount' ) );
+
+		try {
+			/**
+			 * @TODO (sirreal) update to `jetpack-connect` when D11099-code lands
+			 *
+			 * The signup flow is required and affects some post signup activity.
+			 * `account` should be safe until patch lands.
+			 */
+			const { username, bearer_token } = await wpcom.undocumented().usersSocialNew( {
+				...socialInfo,
+				signup_flow_name: 'account' /* 'jetpack-connect' */,
+			} );
+			dispatch( recordTracksEvent( 'calypso_jpc_social_createaccount_success' ) );
+			return { username, bearerToken: bearer_token };
+		} catch ( error ) {
+			const err = {
+				code: error.error,
+				message: error.message,
+				data: error.data,
+			};
+			dispatch(
+				recordTracksEvent( 'calypso_jpc_social_createaccount_error', {
+					error: JSON.stringify( err ),
+					error_code: err.code,
+				} )
+			);
+			throw err;
+		}
+	};
+}
+
+/**
+ * Create a user account
+ *
+ * !! Must have same return shape as createSocialAccount !!
+ *
+ * @param  {Object} userData          …
+ * @param  {Object} userData.username Username
```

Comment:
These types don't seem to be correct.

Reply:
Nice, fixed thm in e2a53bf9fcfa389a037ef6b463d3d2b92823bf46